charmcraft/providers/_multipass.py

```diff
@@ -0,0 +1,175 @@
+            else:
+                logger.debug("Not deleting Multipass VM %r.", name)
+
+        return deleted
+
+    @classmethod
+    def ensure_provider_is_available(cls) -> None:
+        """Ensure provider is available, prompting the user to install it if required.
+
+        :raises CommandError: if provider is not available.
+        """
+        if not multipass.is_installed():
+            if confirm_with_user(
+                "Multipass is required, but not installed. Do you wish to install Multipass "
+                "and configure it with the defaults?",
+                default=False,
+            ):
+                try:
+                    multipass.install()
+                except multipass.MultipassInstallationError as error:
+                    raise CommandError(
+                        "Failed to install Multipass. Visit https://multipass.run/ for "
+                        "instructions on installing Multipass for your operating system."
+                    ) from error
+            else:
+                raise CommandError(
+                    "Multipass is required, but not installed. Visit https://multipass.run/ for "
+                    "instructions on installing Multipass for your operating system."
+                )
+
+        try:
+            multipass.ensure_multipass_is_ready()
+        except multipass.MultipassError as error:
+            raise CommandError(str(error))
+
+    @classmethod
+    def is_provider_available(cls) -> bool:
+        """Check if provider is installed and available for use.
+
+        :returns: True if installed.
+        """
+        return multipass.is_installed()
+
+    @contextlib.contextmanager
+    def launched_environment(
+        self,
+        *,
+        charm_name: str,
+        project_path: pathlib.Path,
+        base: Base,
+        bases_index: int,
+        build_on_index: int,
+    ):
+        """Launch environment for specified base.
+
+        :param charm_name: Name of project.
+        :param project_path: Path to project.
+        :param base: Base to create.
+        :param bases_index: Index of `bases:` entry.
+        :param build_on_index: Index of `build-on` within bases entry.
+        """
+        alias = BASE_CHANNEL_TO_BUILDD_IMAGE_ALIAS[base.channel]
+        target_arch = get_host_architecture()
+
+        instance_name = self.get_instance_name(
+            bases_index=bases_index,
+            build_on_index=build_on_index,
+            project_name=charm_name,
+            project_path=project_path,
+            target_arch=target_arch,
+        )
+
+        environment = self.get_command_environment()
+        base_configuration = CharmcraftBuilddBaseConfiguration(
+            alias=alias, environment=environment, hostname=instance_name
+        )
+        instance = multipass.launch(
+            name=instance_name,
+            base_configuration=base_configuration,
+            image_name=base.channel,
+            cpus=2,
```

Comment:
can you come up with a proposal to customize this?

Reply:
Sure thing.

We can support the options in snap configuration without issue, but as the primary use case for Multipass is not Linux, we should probably support either environment variables (such as CHARMCRAFT_MULTIPASS_MEMORY=4) or command-line parameters (such as --multipass-memory=4).

I assume environment variables would be best?